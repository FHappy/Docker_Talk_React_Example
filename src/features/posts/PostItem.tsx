import React from "react";
import Card from "react-bootstrap/Card";

export interface PostItemProps {
  title: string;
  body: string;
}

const PostItem: React.FC<PostItemProps> = ({ title, body }) => {
  return (
    <Card>
      <Card.Body>
        <Card.Title>{title}</Card.Title>
        <Card.Body>{body}</Card.Body>
      </Card.Body>
    </Card>
  );
};

export default PostItem;
