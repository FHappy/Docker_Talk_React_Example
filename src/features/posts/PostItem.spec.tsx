import { shallow, ShallowWrapper } from "enzyme";
import React from "react";

import PostItem, { PostItemProps } from "./PostItem";

describe("Rendering the PostItem component", () => {
  const mockProps: PostItemProps = {
    title: "Whiskey Business",
    body:
      "Man bun DIY lomo thundercats craft beer actually unicorn gochujang mlkshk la croix fingerstache."
  };

  it("Should render the PostItem component correctly", () => {
    const wrapper: ShallowWrapper = shallow(<PostItem {...mockProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("Should render the PostItem component with the supplied attributes", () => {
    const wrapper: ShallowWrapper = shallow(<PostItem {...mockProps} />);

    const cardTitleHtml = wrapper.find("CardTitle").html();
    const cardBodyHtml = wrapper
      .find("CardBody")
      .last()
      .html();

    expect(cardTitleHtml).toContain(mockProps.title);
    expect(cardBodyHtml).toContain(mockProps.body);
  });
});
