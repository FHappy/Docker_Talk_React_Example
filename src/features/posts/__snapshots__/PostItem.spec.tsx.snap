// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering the PostItem component Should render the PostItem component correctly 1`] = `
<Card
  body={false}
>
  <CardBody>
    <CardTitle>
      Whiskey Business
    </CardTitle>
    <CardBody>
      Man bun DIY lomo thundercats craft beer actually unicorn gochujang mlkshk la croix fingerstache.
    </CardBody>
  </CardBody>
</Card>
`;
